cura/API/Machines.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from typing import Optional, Dict,  List, TYPE_CHECKING, Any
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot, pyqtProperty
+from UM.i18n import i18nCatalog
+from UM.Logger import Logger
+if TYPE_CHECKING:
+    from cura.CuraApplication import CuraApplication
+    from cura.PrinterOutput.PrinterOutputDevice import PrinterOutputDevice
+
+i18n_catalog = i18nCatalog("cura")
+
+##  The account API provides a version-proof bridge to use Ultimaker Accounts
+#
+#   Usage:
+#       ```
+#       from cura.API import CuraAPI
+#       api = CuraAPI()
+#       api.machines.addOutputDeviceToCurrentMachine()
+#       ```
+
+##  Since Cura doesn't have a machine class, we're going to make a fake one to make our lives a
+#   little bit easier.
+class Machine():
+    def __init__(self) -> None:
+        self.hostname = "" # type: str
+        self.group_id = "" # type: str
+        self.group_name = "" # type: str
+        self.um_network_key = "" # type: str
+        self.configuration = {} # type: Dict[str, Any]
+        self.connection_types = [] # type: List[int]
+
+class Machines(QObject):
+
+    def __init__(self, application: "CuraApplication", parent = None) -> None:
+        super().__init__(parent)
+        self._application = application
```

Comment:
This is the only way I know to avoid a circular dependency, and was done already by Cura team.

Reply:
Yes, whenever I see this being done by someone in the Cura team or some external contributor, I'm also asking them to change it.

Reply:
Ok but then you guys can figure out how to avoid that circular dependency and in the mean time i'm going to copy the established solution that is in the other API endpoints

Reply:
You can work around this circular dependency by calling `#import cura.CuraApplication` instead of `from cura.CuraApplication import CuraApplication`. In general it's strange that you'd have a circular dependency with CuraApplication from a plug-in though, because that would mean that importing CuraApplication somehow imports your plug-in?

Reply:
No, it's a circular dependency with Cura importing the API module (presumably for the `.getCuraAPI()` method). Lipu made a commit caching the app instance with the description "Fix circular dependency" or something along those lines. I copied that pattern when extending it.